src/vmm/src/vstate/vm.rs

```diff
@@ -183,15 +189,32 @@ impl Vm {
             Ok(Vm {
                 fd: vm_fd,
                 max_memslots,
+                kvm_cap_modifiers,
                 supported_cpuid,
                 msrs_to_save,
             })
         }
     }
 
-    fn check_capabilities(kvm: &Kvm, capabilities: &[Cap]) -> Result<(), VmError> {
+    fn combine_capabilities(kvm_cap_modifiers: &[KvmCapability]) -> Vec<u32> {
+        let mut total_caps = Self::DEFAULT_CAPABILITIES.to_vec();
+        for modifier in kvm_cap_modifiers.iter() {
+            match modifier {
+                KvmCapability::Add(cap) => total_caps.push(*cap),
+                KvmCapability::Remove(cap) => {
+                    if let Some(pos) = total_caps.iter().position(|def_cap| def_cap == cap) {
```

Comment:
it looks like we only do this check when we remove. Do we not need a similar check when we add in order to avoid duplicates?

Reply:
I guess I can add it. But ultimately it is not an error to check for same capability multiple times.

Reply:
Added the check